jabgui/src/main/java/org/jabref/gui/walkthrough/SingleWindowOverlay.java

```diff
@@ -0,0 +1,308 @@
+        overlayPane.setAlignment(Pos.TOP_LEFT);
+        overlayPane.getChildren().add(tooltipContainer);
+
+        setupClipping(tooltipContainer);
+
+        ChangeListener<Bounds> layoutListener = (_, _, newBounds) -> {
+            if (newBounds.getWidth() > 0 && newBounds.getHeight() > 0) {
+                positionTooltipWithArrow(tooltipContainer, content, arrow, targetNode, step);
+            }
+        };
+        content.boundsInLocalProperty().addListener(layoutListener);
+
+        if (content.getBoundsInLocal().getWidth() > 0) {
+            positionTooltipWithArrow(tooltipContainer, content, arrow, targetNode, step);
+        }
+
+        cleanUpTasks.add(() -> content.boundsInLocalProperty().removeListener(layoutListener));
+    }
+
+    private void displayPanelContent(Node content, Pos position) {
+        overlayPane.getChildren().clear();
+        overlayPane.getChildren().add(content);
+
+        setupClipping(content);
+
+        overlayPane.getRowConstraints().clear();
+        overlayPane.getColumnConstraints().clear();
+        overlayPane.setAlignment(position);
+
+        GridPane.setHgrow(content, Priority.NEVER);
+        GridPane.setVgrow(content, Priority.NEVER);
+        GridPane.setFillWidth(content, false);
+        GridPane.setFillHeight(content, false);
+
+        RowConstraints rowConstraints = new RowConstraints();
+        ColumnConstraints columnConstraints = new ColumnConstraints();
+
+        switch (position) {
+            case CENTER_LEFT:
+            case CENTER_RIGHT:
+                rowConstraints.setVgrow(Priority.ALWAYS);
+                columnConstraints.setHgrow(Priority.NEVER);
+                GridPane.setFillHeight(content, true);
+                break;
+            case TOP_CENTER:
+            case BOTTOM_CENTER:
+                columnConstraints.setHgrow(Priority.ALWAYS);
+                rowConstraints.setVgrow(Priority.NEVER);
+                GridPane.setFillWidth(content, true);
+                break;
+            default:
+                LOGGER.warn("Unsupported position for panel step: {}", position);
+                break;
+        }
+
+        overlayPane.getRowConstraints().add(rowConstraints);
+        overlayPane.getColumnConstraints().add(columnConstraints);
+    }
+
+    private void setupClipping(Node node) {
+        ChangeListener<? super Bounds> listener = (_, _, bounds) -> {
+            Rectangle clip = new Rectangle(bounds.getMinX(), bounds.getMinY(), bounds.getWidth(), bounds.getHeight());
+            overlayPane.setClip(clip);
+        };
+        node.boundsInParentProperty().addListener(listener);
+        cleanUpTasks.add(() -> node.boundsInParentProperty().removeListener(listener));
+        cleanUpTasks.add(() -> overlayPane.setClip(null));
+    }
+
+    private Polygon createArrow() {
+        Polygon arrow = new Polygon();
+        arrow.getPoints().addAll(0.0, 0.0,
+                12.0, 15.0,
+                -12.0, 15.0);
```

Comment:
put into base.css
See checks in checkboxes in modena.css

Reply:
There is already a styled control in controlsfx.
https://controlsfx.github.io/features/popover/
@Siedlerchr is one of the maintainers